dist/aurelia-dependency-injection.d.ts

```diff
@@ -1,185 +1,209 @@
+export interface DependencyCtor<TBase, TImpl extends TBase, TArgs = any> {
+  new(...args: TArgs[]): TImpl & TBase;
+}
+
+export type DependencyCtorKeyOrNever<TBase, TImpl, TArgs = any> = TImpl extends TBase ? DependencyCtor<TBase, TImpl, TArgs> : string;
+export type StringOrNever<TBase, TImpl, TArgs = any> = TBase extends string ? TBase : never;
```

Comment:
is `TImpl` needed?

Reply:
Yeah, It's needed for sub classes because the return of the `new()` needs to be a generic.
e.g. https://github.com/DrSammyD/dependency-injection/blob/1f7cec4667473f15e889d3d2e9a737c98248fa85/test/container.spec.ts#L232